fluss-client/src/main/java/com/alibaba/fluss/client/lookup/PrefixKeyLookuper.java

```diff
@@ -80,15 +86,19 @@ public PrefixKeyLookuper(
         this.lookupClient = lookupClient;
         // the row type of the input lookup row
         RowType lookupRowType = tableInfo.getRowType().project(lookupColumnNames);
-        this.bucketKeyEncoder =
-                KeyEncoder.createKeyEncoder(lookupRowType, tableInfo.getBucketKeys());
-
-        if (tableInfo.getTableConfig().isDataLakeEnabled()) {
-            this.lakeTableBucketAssigner =
-                    new LakeTableBucketAssigner(
-                            lookupRowType, tableInfo.getBucketKeys(), numBuckets);
+        Optional<DataLakeFormat> optDataLakeFormat = tableInfo.getTableConfig().getDataLakeFormat();
+
+        if (optDataLakeFormat.isPresent()) {
+            DataLakeFormat dataLakeFormat = optDataLakeFormat.get();
+            this.bucketKeyEncoder =
+                    LakeKeyEncoderFactory.createKeyEncoder(
+                            dataLakeFormat, lookupRowType, tableInfo.getBucketKeys());
+            this.lakeBucketAssigner =
+                    LakeBucketAssignerFactory.createLakeBucketAssigner(dataLakeFormat, numBuckets);
         } else {
-            this.lakeTableBucketAssigner = null;
+            this.bucketKeyEncoder =
+                    CompactedKeyEncoder.createKeyEncoder(lookupRowType, tableInfo.getBucketKeys());
+            this.lakeBucketAssigner = null;
```

Comment:
Constructing the `bucketKeyEncoder` is still very complex. However, we can unify the default and lake assigners into one util method.